### PR TITLE
use gcc 7 on xenial, and link libstdc++ statically on xenial

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
tested at https://shaman.ceph.com/builds/ceph/wip-gcc-7-on-xenial/af46f642601ecdd9e70e789e7bf48ac984b22bc3/

File: ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml [hunk 1]
```diff
@@ -47,6 +47,7 @@
                   predefined-parameters: |
                     BRANCH=${GIT_BRANCH}
                     FORCE=True
+                    FLAVOR=notcmalloc
```
Comment (alfredodeza) [inline ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml]: I think these `FLAVOR` additions just got merged with a13e43401dff61ec5d0e705d76bde70ead72a900 right?

Comment (tchaikov) [reply ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml]: @alfredodeza yes. the github's merge button takes this into consideration. so this PR is now equivalent to bd61b62 + a8a926d.

Comment (tchaikov) [reply ceph-dev-new-trigger/config/definitions/ceph-dev-new-trigger.yml]: okay, rebased and repushed.

**Comment by ktdreyer** [comment]
Why are we using apt with `--allow-unauthenticated` ?

**Comment by tchaikov** [comment]
@ktdreyer because i am too lazy to import the PPA's signing key to the APT keyring.

**Comment by liewegas** [comment]
let's fix that in a follow-up pr?

**Comment by tchaikov** [comment]
#939

**Pull request merged**